Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -497,18 +497,24 @@ public typealias StartPurchaseBlock = (@escaping PurchaseCompletedBlock) -> Void
         Self.purchases.value = nil
     }
 
-    static func setDefaultInstance(_ purchases: Purchases) {
-        self.purchases.modify { currentInstance in
+    @discardableResult
+    static func setDefaultInstance(_ purchases: @autoclosure () -> Purchases) -> Purchases {
```

Comment:
Do you mind adding a comment here why this is an `autoclosure` and not just an instance of `Purchases`?

Reply:
Good idea